BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m

```diff
@@ -315,67 +316,23 @@ - (void)connection:(NSURLConnection *)connection willSendRequestForAuthenticatio
-					requestUserConfirmation = YES;
-					break;
-				case kSecTrustResultFatalTrustFailure:
-					// Trust denied; no simple fix is available. For example, if a certificate cannot be verified because it is corrupted, trust cannot be established without replacing the certificate. This value may be returned by the SecTrustEvaluate function but not stored as part of the user trust settings.
-					requestUserConfirmation = YES;
-					break;
-				case kSecTrustResultOtherError:
-					// A failure other than that of trust evaluation; for example, an internal failure of the SecTrustEvaluate function. This value may be returned by the SecTrustEvaluate function but not stored as part of the user trust settings.
-					requestUserConfirmation = YES;
-					break;
-				default:
-					break;
-			}
-		}
-		else
-		{
-			// The SecTrustEvaluate method failed
-			BOXLog(@"Sec trust evaluate failed to establish a value, so prompt the user.");
-			requestUserConfirmation = YES;
-		}
+        // By default, allow a certificate if its status was evaluated successfully and the result is
+        // that it should be trusted
+        BOOL shouldTrustServer = (status == errSecSuccess && (trustResult == kSecTrustResultProceed || trustResult == kSecTrustResultUnspecified));
 
-		if (requestUserConfirmation)
+		if (shouldTrustServer)
 		{
-			self.authenticationChallenge = challenge;
-			UIAlertView *serverTrustAlertView = [[UIAlertView alloc] initWithTitle:NSLocalizedString(@"Cannot Verify Server Identity", @"Alert view title: title for SSO server trust challenge")
-																		   message:[NSString stringWithFormat:NSLocalizedString(@"Box cannot verify the identity of \"%@\". Would you like to continue anyway?", @"Alert view message: Message for SSO server trust challenge, giving the user the host of the server who's identity cannot be verified."), [[challenge protectionSpace] host]]
-																		  delegate:self
-																 cancelButtonTitle:NSLocalizedString(@"Cancel", @"Label: Cancel action. Usually used on buttons.")
-																 otherButtonTitles:NSLocalizedString(@"Continue", @"Label: Continue action. Usually used on buttons."), nil];
-			serverTrustAlertView.tag = BOX_SSO_SERVER_TRUST_ALERT_TAG;
-			[serverTrustAlertView show];
-		}
+            [self completeServerTrustAuthenticationChallenge:challenge shouldTrust:shouldTrustServer];
+        }
 		else
 		{
-			// By default, allow a certificate if its status was evaluated successfully and the result is
-			// that it should be trusted
-			BOOL shouldTrustServer = (status == errSecSuccess && (trustResult == kSecTrustResultProceed || trustResult == kSecTrustResultUnspecified));
-			[self completeServerTrustAuthenticationChallenge:challenge shouldTrust:shouldTrustServer];
+            UIAlertView *serverTrustAlertView = [[UIAlertView alloc] initWithTitle:NSLocalizedString(@"Cannot Verify Server Identity", @"Alert view title: title for SSO server trust challenge")
```

Comment:
If we want to handle the failure case here directly, we should clean up the completeServerTrust method - it no longer needs a shouldTrust property, since we're only using it when the host is trusted, and we can take out the conditional in the implementation.